tests/Schema/TableTest.php

```diff
@@ -1755,4 +1759,121 @@ public function testInvalidPrimaryKeyIndex(): void
         $this->expectException(InvalidState::class);
         $table->getPrimaryKeyConstraint();
     }
+
+    public function testAutoGeneratedForeignKeyNamesForConflictingReferences(): void
+    {
+        $table = Table::editor()
+            ->setUnquotedName('t')
+            ->setColumns(
+                Column::editor()
+                    ->setUnquotedName('id')
+                    ->setTypeName(Types::INTEGER)
+                    ->create(),
+                Column::editor()
+                    ->setUnquotedName('t_id')
+                    ->setTypeName(Types::INTEGER)
+                    ->create(),
+            )
+            ->create();
+
+        // FK 1: should be added
+        $table->addForeignKeyConstraint('t1', ['t_id'], ['id']);
+        // FK 2: should be added
+        $table->addForeignKeyConstraint('t2', ['t_id'], ['id']);
+        // FK 3: should be added
+        $table->addForeignKeyConstraint('t2', ['t_id'], ['second_id']);
+        // FK 4: duplicate of FK 2 – should NOT be added
+        $table->addForeignKeyConstraint('t2', ['t_id'], ['id']);
+        // FK 5: duplicate of FK 1 – should NOT be added
+        $table->addForeignKeyConstraint('t1', ['t_id'], ['id']);
```

Comment:
Please keep it to the necessary minimum. It doesn't take 5 constraints to verify a single condition.

Ideally, for each code path we could have a single test with a data provider. The provider would provide some input (e.g. the constraint name (or names), or definition (or definitions) and a boolean that defines whether this input should trigger a deprecation).

The test would apply this input and assert whether the deprecation was or was not triggered. This way, the test will ensure that this is exactly the difference between the inputs (e.g. the presence of a duplicate or its absence) causes triggering a deprecation.